src/m365/pp/commands/website/website-weblink-list.ts

```diff
@@ -0,0 +1,127 @@
+import { Logger } from '../../../../cli/Logger.js';
+import { globalOptionsZod } from '../../../../Command.js';
+import { z } from 'zod';
+import { zod } from '../../../../utils/zod.js';
+import request, { CliRequestOptions } from '../../../../request.js';
+import { odata } from '../../../../utils/odata.js';
+import { powerPlatform } from '../../../../utils/powerPlatform.js';
+import { validation } from '../../../../utils/validation.js';
+import PowerPlatformCommand from '../../../base/PowerPlatformCommand.js';
+import commands from '../../commands.js';
+import { PpWebSiteOptions } from '../Website.js';
+
+
+
+export const options = globalOptionsZod
+  .extend({
+    environmentName: zod.alias('e', z.string()),
+    websiteId: z.string().refine(id => validation.isValidGuid(id) === true, id => ({ message: `${id} is not a valid GUID` })).optional(),
+    websiteName: z.string().optional(),
+    asAdmin: z.boolean().optional()
+  })
+  .refine(options => !(options.websiteId !== undefined && options.websiteName !== undefined), {
+    message: `Either websiteId or websiteName is required, but not both.`
+  })
+  .refine(options => !(options.websiteId === undefined && options.websiteName === undefined), {
+    message: `Either websiteId or websiteName is required.`
+  });
+
+declare type Options = z.infer<typeof options>;
+
+interface CommandArgs {
+  options: Options;
+}
+
+class PpWebSiteWebLinkListCommand extends PowerPlatformCommand {
+  public get name(): string {
+    return commands.WEBSITE_WEBLINK_LIST;
+  }
+
+  public get description(): string {
+    return 'List all weblinks for the specified Power Pages website';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['mspp_name', 'mspp_weblinkid', 'mspp_description', 'statecode'];
+  }
+
+  public get schema(): z.ZodTypeAny {
+    return options;
+  }
+
+  private async getWebSiteId(dynamicsApiUrl: string, args: CommandArgs): Promise<any> {
+    if (args.options.websiteId) {
+      return args.options.websiteId;
+    }
+    const options: PpWebSiteOptions = {
+      environmentName: args.options.environmentName,
+      id: args.options.websiteId,
+      name: args.options.websiteName,
+      output: 'json'
+    };
+
+    const requestOptions: CliRequestOptions = {
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    if (options.name) {
```

Comment:
What's this options.name doing here? I think you should not need to check if the website name has a value. Its an option set with websiteId, and we've already ascertained in the validation  that a value should be there in one of the options.